R/literal_coercion_linter.R

```diff
@@ -12,23 +12,31 @@
 #' @seealso [linters] for a complete list of linters available in lintr.
 #' @export
 literal_coercion_linter <- function() {
-  coercers <- xp_text_in_table(paste0(
-    "as.",
-    c("logical", "integer", "numeric", "double", "character")
-  ))
+  rlang_coercers <- xp_text_in_table(
+    c("lgl", "int", "dbl", "chr")
+  )
+  coercers <- xp_text_in_table(
+      paste0("as.", c("logical", "integer", "numeric", "double", "character"))
+  )
   # notes for clarification:
   #  - as.integer(1e6) is arguably easier to read than 1000000L
   #  - in x$"abc", the "abc" STR_CONST is at the top level, so exclude OP-DOLLAR
   #  - need condition against STR_CONST w/ EQ_SUB to skip quoted keyword arguments (see tests)
   xpath <- glue::glue("//expr[
-    expr[1][SYMBOL_FUNCTION_CALL[ {coercers} ]]
-    and expr[2][
-      not(OP-DOLLAR)
-      and (
-        NUM_CONST[not(contains(translate(text(), 'E', 'e'), 'e'))]
-        or STR_CONST[not(following-sibling::*[1][self::EQ_SUB])]
-      )
-    ]
+    (
+      expr[1][SYMBOL_FUNCTION_CALL[ {coercers} ]]
+      and expr[2][
+        not(OP-DOLLAR)
+        and (
+          NUM_CONST[not(contains(translate(text(), 'E', 'e'), 'e'))]
+          or STR_CONST[not(following-sibling::*[1][self::EQ_SUB])]
+        )
+      ]
+    ) or
+    (
+      expr[1][SYMBOL_FUNCTION_CALL[ {rlang_coercers} ]]
+      and not(OP-COMMA)
```

Comment:
does rlant support list2 construction, i.e., does `lgl(1,)` work? if so, may need count(expr) here instead

Reply:
Good catch! I have changed xpath to catch/lint this possibility as well.